sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertIntoHiveTableSuite.scala

```diff
@@ -212,4 +214,77 @@ class InsertIntoHiveTableSuite extends QueryTest with TestHiveSingleton with Bef
 
     sql("DROP TABLE hiveTableWithStructValue")
   }
+
+  test("Reject partitioning that does not match table") {
+    withSQLConf(("hive.exec.dynamic.partition.mode", "nonstrict")) {
```

Comment:
I doubt whether this configuration is effective/respected in Spark. Not quite familiar with the newly refactored Spark configuration API. I guess it's probably not respected, and you don't need it to create tables partitioned by dynamic partitions only in Spark.